wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -231,7 +255,11 @@ private CodeBlock singleAdapterFor(ProtoType type) {
     } else if (type.isMap()) {
       throw new IllegalArgumentException("Cannot create single adapter for map type " + type);
     } else {
-      result.add("$T.ADAPTER", typeName(type));
+      if (protoTypeToAdapter.containsKey(type)) {
```

Comment:
By the way, this is a bad antipattern because you’re doing 2 map lookups. Much more efficient to do a single map lookup and check for null, like so:

```
ClassName className = protoTypeToAdapter.get(type);
if (className != null) {
  result.add("$T.ADAPTER", className);
} else {
...
```
```